src/view_translate_bisulfite.hpp

```diff
@@ -0,0 +1,428 @@
+// together with this file.
+//
+// Lambda is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+//
+// ==========================================================================
+// view_translate_bisulfite.hpp: View that converts dna sequences into bisulfite
+// forward and reverse sequences
+// ==========================================================================
+
+#pragma once
+
+#include <seqan3/alphabet/composite/semialphabet_any.hpp>
+#include <seqan3/alphabet/nucleotide/concept.hpp>
+#include <seqan3/alphabet/nucleotide/all.hpp>
+#include <seqan3/core/type_traits/range.hpp>
+#include <seqan3/range/detail/random_access_iterator.hpp>
+#include <seqan3/range/views/deep.hpp>
+#include <seqan3/range/views/complement.hpp>
+#include <seqan3/range/views/to.hpp>
+#include <seqan3/range/concept.hpp>
+#include <seqan3/std/ranges>
+#include <seqan3/std/concepts>
+#include "view_pos_transform.hpp"
+
+// Rank
+// 0 = A forward
+// 1 = C/T forward
+// 2 = G forward
+// 3 = A/G reverse
+// 4 = C reverse
+// 5 = T reverse
+
+static const std::array<uint8_t, 4> dna4_to_rank_bs_fwd = {0, 1, 2, 1};
+static const std::array<uint8_t, 4> dna4_to_rank_bs_rev = {3, 4, 3, 5};
+
+struct to_bisulfite_semialphabet_fn
+{
+    constexpr auto operator()(bool const fwd) const
+    {
+        return seqan3::detail::adaptor_from_functor{*this, fwd};
+    }
+
+    template <std::ranges::range urng_t>
+    constexpr auto operator()(urng_t && urange, bool const fwd) const
+    {
+        static_assert(std::ranges::viewable_range<urng_t>,
+            "The range parameter to views::to_bisulfite_semialphabet cannot be a temporary of a non-view range.");
+        static_assert(std::ranges::sized_range<urng_t>,
+            "The range parameter to views::to_bisulfite_semialphabet must model std::ranges::sized_range.");
+        static_assert(std::ranges::random_access_range<urng_t>,
+            "The range parameter to views::to_bisulfite_semialphabet must model std::ranges::random_access_range.");
+        static_assert(std::is_same_v<std::remove_const_t<seqan3::innermost_value_type_t<urng_t>>, seqan3::dna4>,
+            "The range parameter to views::to_bisulfite_semialphabet must be over elements of seqan3::dna4.");
+
+        return std::forward<urng_t>(urange) | std::views::transform([fwd] (auto const in)
+        {
+            if (fwd)
+            {
+                return seqan3::assign_rank_to(dna4_to_rank_bs_fwd[seqan3::to_rank(in)], seqan3::semialphabet_any<6>{});
+            }
+            else
+            {
+                return seqan3::assign_rank_to(dna4_to_rank_bs_rev[seqan3::to_rank(in)], seqan3::semialphabet_any<6>{});
+            }
+        });
+    }
+};
+
+namespace seqan3::views
+{
+
+inline constexpr auto to_bisulfite_semialphabet = seqan3::views::deep{to_bisulfite_semialphabet_fn{}};
```

Comment:
We don't need full Doxygen, but maybe a few comment lines per view what it does would be nice?